pkg/cli/admin/release/new.go

```diff
@@ -810,6 +810,24 @@ func resolveImageStreamTagsToReferenceMode(inputIS, is *imageapi.ImageStream, re
 				continue
 			}
 
+			statusRef := findStatusTagEvents(inputIS.Status.Tags, ref.Name)
```

Comment:
Nit: `imageutil.LatestTaggedImage` does just that.

Reply:
I'll open followup to change these, I see similarly we have methods searching through spec.

Reply:
It's not quite the same though iirc.

Reply:
IT's not the same, we need generation here.

Reply:
Both are identical, both iterate over `.status.tags` and you're getting a `TagEvent` which contains the generation. We don't do more comparison than just simple iteration and name matching. After which we in both cases pick the 0th element from tag events.